app/gen-server/lib/HomeDBManager.ts

```diff
@@ -2479,6 +2490,85 @@ export class HomeDBManager extends EventEmitter {
       .getOne() || undefined;
   }
 
+  // can be factorized with getShareByLinkId
+  public async getDocApiKeyByLinkId(docId: string, linkId: string): Promise<Share | undefined> {
+    return await this._connection.createQueryBuilder()
+      .select('shares')
+      .from(Share, 'shares')
+      .where('shares.doc_id = :docId', {docId})
+        .andWhere('shares.link_id = :linkId', {linkId})
+      .getOne() || undefined;
+  }
+
+  public async getDocApiKeyByKey(key: string): Promise<Share | null> {
+    return await this.getShareByKey(key);
+  }
```

Comment:
Nitpicking: you may remove the `async` and the `await` here. The fact that you return a promise should be enough.

Reply:
Done